### PR TITLE
Change the way to define lists (fixes #217)

### LINKED ISSUE #217: List item type definition is cumbersome
https://github.com/n1k0/kept/pull/25/files#r16965400

### PR DESCRIPTION
**Pull request by leplatrem**
See #217 

Usage is easier. Code is shorter. What else?™

### CODE REVIEW

File: daybed/tests/test_schemas_list.py [hunk 1]
```diff
@@ -18,15 +19,18 @@ class ListFieldTest(unittest.TestCase):
...
     def test_can_be_defined_with_a_field_type(self):
         field = self.schema.deserialize(self.definition)
+        for defaultattr in ['required', 'label']:
+            field.pop(defaultattr)
+            field['item'].pop(defaultattr)
         self.assertDictEqual(self.definition, field)
 
     def test_is_not_valid_if_field_type_unknown(self):
-        self.definition['itemtype'] = 'asteroid'
+        self.definition['item']['type'] = 'asteroid'
```
Comment (almet) [inline daybed/tests/test_schemas_list.py:33]: `']['` ftw.

**Comment by almet** [comment]
Sounds good to me.

This would need to have the documentation updated as well.

File: daybed/schemas/list.py [hunk 2]
```diff
@@ -38,20 +23,13 @@ class ListField(TypeField):
...
         schema = super(ListField, cls).definition(**kwargs)
-        schema.add(SchemaNode(String(),
-                              name='itemtype',
-                              validator=OneOf(registry.names),
-                              missing=drop))
-        schema.add(SchemaNode(Mapping(unknown='preserve'),
-                              name='parameters',
-                              missing=drop))
-        schema.validator = ParametersMatchItemType()
+        schema.add(SchemaNode(TypeFieldNode(named=False),
```
Comment (almet) [inline daybed/schemas/list.py:26]: Okay, so it's only named when it's the top level object?

Comment (leplatrem) [inline daybed/schemas/list.py:26]: Yes, it's only named when used in a definition fields list

**Comment by leplatrem** [comment]
Which documentation ? :)

**Comment by almet** [comment]
We should have some doc about all the existing types and how to use them.

**Pull request merged**